src/ServiceControl.Plugin.Nsb6.SagaAudit/CaptureSagaResultingMessagesBehavior.cs

```diff
@@ -1,68 +1,61 @@
+            var logicalMessage = context.Message;
             if (logicalMessage == null)
             {
                 //this can happen on control messages
                 return;
             }
 
-            var sendOptions = context.DeliveryOptions as SendOptions;
-            if (sendOptions != null)
+            TimeSpan? deliveryDelay = null;
+            DelayDeliveryWith delayDeliveryWith;
+            if (context.Extensions.TryGetDeliveryConstraint(out delayDeliveryWith))
             {
-                var sagaResultingMessage = new SagaChangeOutput
-                {
-                    
-                    ResultingMessageId = context.OutgoingMessage.Id,
-                    TimeSent = DateTimeExtensions.ToUtcDateTime(context.OutgoingMessage.Headers[Headers.TimeSent]),
-                    MessageType = logicalMessage.MessageType.ToString(),
-                    DeliveryDelay = sendOptions.DelayDeliveryWith,
-                    DeliveryAt = sendOptions.DeliverAt,
-                    Destination = sendOptions.Destination.ToString(),
-                    Intent = "Send" //TODO: How to get the proper message intent!?
-                };
-                sagaUpdatedMessage.ResultingMessages.Add(sagaResultingMessage);
+                deliveryDelay = delayDeliveryWith.Delay;
             }
 
-            if (context.DeliveryOptions is PublishOptions)
+            DateTime? doNotDeliverBefore = null;
+            DoNotDeliverBefore notDeliverBefore;
+            if (context.Extensions.TryGetDeliveryConstraint(out notDeliverBefore))
             {
-                var sagaResultingMessage = new SagaChangeOutput
-                {
-                    ResultingMessageId = context.OutgoingMessage.Id,
-                    TimeSent = DateTimeExtensions.ToUtcDateTime(context.OutgoingMessage.Headers[Headers.TimeSent]),
-                    MessageType = logicalMessage.MessageType.ToString(),
-                    //TODO: Can we remove the DeliveryDelay and DeliveryAt here??
-                    //DeliveryDelay = publishOptions.DelayDeliveryWith,
-                    //DeliveryAt = publishOptions.DeliverAt,
-                    //Destination = GetDestination(context),
-                    Intent = "Publish" //TODO: get the message intent the right way!
-                };
-                sagaUpdatedMessage.ResultingMessages.Add(sagaResultingMessage);
+                doNotDeliverBefore = notDeliverBefore.At;
             }
+
+            var sagaResultingMessage = new SagaChangeOutput
+            {
+                ResultingMessageId = context.MessageId,
+                TimeSent = DateTimeExtensions.ToUtcDateTime(context.Headers[Headers.TimeSent]),
+                MessageType = logicalMessage.MessageType.ToString(),
+                DeliveryDelay = deliveryDelay,
+                DeliveryAt = doNotDeliverBefore,
+                Destination = string.Empty /* TODO: How to get this properly, maybe hook in later in the pipeline?*/,
```

Comment:
@SzymonPobiega @janovesk Any ideas to properly get the destination for UnicastRoutings? Or should we deprecate the destination? // cc @Particular/servicecontrol-maintainers